### PR TITLE
Catch module theme installation error, add error message for invalid module

### PR DESCRIPTION
**Pull request by Quetzacoalt91**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | 1.7.4.x
| Description?  | <ul><li>If a module requested by a theme could not be installed, no error was thrown and the execution was going on. Enabling the same uninstalled module was failing and the message wasn't explicit enough.</li><li>A new error message has finally been added for invalid modules (missing or with a syntax error). Merchants will now get a specific error instead of the current "no more details provided" :raised_hands: )</li></ul>
| Type?         | bug fix
| Category?     | BO
| BC breaks?    | Nope
| Deprecations? | Nope
| Fixed ticket? | http://forge.prestashop.com/browse/BOOM-5480
| How to test?  | Install the theme provided in the forge issue

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/9089)

### CODE REVIEW

File: src/Core/Addon/Module/ModuleManager.php [hunk 1]
```diff
@@ -590,6 +590,11 @@ public function getError($name)
         if ($module->hasValidInstance()) {
             $errors = $module->getInstance()->getErrors();
             $message = array_pop($errors);
+        } else {
+            // Invalid instance: Missing or with syntax error
+            $message = $this->translator->trans('The module is invalid and cannot be loaded.',
```
Comment (PierreRambaud) [inline src/Core/Addon/Module/ModuleManager.php]: Can you split arguments across multiple lines, it's easier to read:
```php
$message = $this->translator->trans(
    'The module is invalid and cannot be loaded.',
    [],
    'Admin.Modules.Notification'
);
```

File: src/Core/Addon/Theme/ThemeManager.php [hunk 2]
```diff
@@ -284,8 +284,17 @@ private function doEnableModules(array $modules)
...
+                && !$moduleManager->install($moduleName)) {
+                throw new PrestaShopException(
+                    $this->translator->trans('Cannot %action% module %module%. %error_details%',
+                    array(
+                        '%action%' => 'install',
+                        '%module%' => $moduleName,
+                        '%error_details%' => $moduleManager->getError($moduleName),
+                    ),
+                    'Admin.Modules.Notification')
+                );
```
Comment (PierreRambaud) [inline src/Core/Addon/Theme/ThemeManager.php]: Same here please:

```php
if (!$moduleManager->isInstalled($moduleName)
    && !$moduleManager->install($moduleName)
) {
    throw new PrestaShopException(
        $this->translator->trans(
            'Cannot %action% module %module%. %error_details%',
            [
                '%action%' => 'install',
                '%module%' => $moduleName,
                '%error_details%' => $moduleManager->getError($moduleName),
            ],
            'Admin.Modules.Notification'
        )
    );
}
```

**Review by PierreRambaud — APPROVED** [review]

File: src/Core/Addon/Module/ModuleManager.php [hunk 3]
```diff
@@ -348,8 +352,10 @@ public function disable($name)
         if (!$this->adminModuleProvider->isAllowedAccess(__FUNCTION__, $name)) {
```
Comment (mickaelandrieu) [inline src/Core/Addon/Module/ModuleManager.php:352]: sounds like this block of code is used multiple times, how about creating a reusable function?

Comment (Quetzacoalt91) [reply src/Core/Addon/Module/ModuleManager.php:352]: This is already a reused function, what do you need more?

Comment (mickaelandrieu) [reply src/Core/Addon/Module/ModuleManager.php:352]: the `if` block could be one function too: how about `throwIfModuleAccessNotAllowed($name)`?

Comment (PierreRambaud) [reply src/Core/Addon/Module/ModuleManager.php:352]: Maybe overkill, more you must pass the `__FUNCTION__`, `$name`, translated message, or message, parameters and domain. :/

Comment (Quetzacoalt91) [reply src/Core/Addon/Module/ModuleManager.php:352]: Sounds better to directly send the result of `$this->translator->trans(...)` in that case.

Comment (Quetzacoalt91) [reply src/Core/Addon/Module/ModuleManager.php:352]: Do you really expect another refacto in this PR?
As I'm changing some strings, I'll have to wait for 1.7.5.0 to make them available if we don't accept it for the next minor release.

Comment (mickaelandrieu) [reply src/Core/Addon/Module/ModuleManager.php:352]: nope, it's is a suggestion :+1: @PierreRambaud have forgotten to add the "waiting for QA" label

**Comment by eternoendless** [comment]
Thank you @Quetzacoalt91

**Pull request merged**